File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n=len(nums)
        dp=[0 for _ in range(n)]

        def houseRob(n):
            if n<0 :
                return 0
            if n==0:
                return nums[n]

            if dp[n]:
                return dp[n]

            dp[n]=  max(nums[n]+houseRob(n-2),houseRob(n-1))
            return dp[n]

        return houseRob(n-1)

File: Data Structures & Algorithms/maximum-subarray/submission-2.java
class Solution {
    public int maxSubArray(int[] nums) {
        int temp=0;
        int result=nums[0];
        for(int i=0;i<nums.length;i++)
        {
            temp+=nums[i];
            result=Math.max(temp,result);
            if(temp<0)
            {
                temp=0;
            }
        }
        return result;
    }
}